### ISSUE TITLE
Strange test failure related to CircleCI caching

### ISSUE THREAD

**Issue by stefsmeets** [ESMValGroup/ESMValCore#952]
**Describe the bug**
I have had a whole slew of issues in #907 that I'm currently working on. I merged master to resolve some issues with the CI. This introduced some errors with CircleCi that have nothing to do with my PR. I think I finally tracked down the bug to be related to the caching in CircleCI.

Currently, we cache to the `test-{{ .Branch }}` key in `.circleci/config.yml`. In my case, the cache was not updated after the build system was changed by merging master (something to do with cartopy/numpy). This means all tests now fail, and no way to resolve the issue as the cache is only updated after a successful run.

Unfortunately, CircleCi has no way to clear the cache, the only way is to update the key.
I suggest we link the cache key to `setup.py` to avoid such issues in the future. This is also one of the solutions proposed in the CircleCI docs: https://support.circleci.com/hc/en-us/articles/115015426888-Clear-project-dependency-cache

I'm currently testing if this resolves the issue in #907.

**Comment by stefsmeets** [comment]
I just noticed #951, which may have been part of the problem while I was trying to debug this...

**Comment by stefsmeets** [comment]
OK, it seems like changing the CircleCI cache did nothing to solve my issue.

I keep getting this, which I cannot reproduce locally:
```
self = <tests.unit.preprocessor._regrid.test_regrid.Test testMethod=test_regrid__cell_specification>

    def test_regrid__cell_specification(self):
        specs = ['1x1', '2x2', '3x3', '4x4', '5x5']
        scheme = 'linear'
        for spec in specs:
            result = regrid(self.src_cube, spec, scheme)
            self.assertEqual(result, self.regridded_cube)
>           self._check(spec, scheme, spec=True)

tests/unit/preprocessor/_regrid/test_regrid.py:107: 
_ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ _ 
tests/unit/preprocessor/_regrid/test_regrid.py:23: in _check
    self.assertEqual(_CACHE[spec], self.tgt_grid)
E   AssertionError: <iris 'Cube' of unknown / (unknown) (latitude: 180; longitude: 360)> != <Mock spec='Cube' id='140424322565072'>
```

**Comment by stefsmeets** [comment]
It probably is not related to CircleCI. I'm currently thinking that the implementation of the regrid `_CACHE` is being filled with ambiguous keys, i.e. after multiple calls with cubes from different sources, there may be a name collision, since it just retrieves what is there for `1x1`, `2x2`, etc.

It just happens that my PR, which has some tests with regridding, triggers this behaviour.

**Comment by stefsmeets** [comment]
This can be closed, I created a new issue in #953 describing the cache retrieval bug.
